lib/src/int64.dart

```diff
@@ -672,18 +750,34 @@ class Int64 implements IntX {
   @override
   String toString() => _toRadixString(10);
 
-  // TODO(rice) - Make this faster by avoiding arithmetic.
+  /// Reusable buffer used by [toHexString].
+  static Uint8List? _buffer;
+
   @override
   String toHexString() {
     if (isZero) return '0';
-    Int64 x = this;
-    String hexStr = '';
-    while (!x.isZero) {
-      int digit = x._l & 0xf;
-      hexStr = '${_hexDigit(digit)}$hexStr';
-      x = x.shiftRightUnsigned(4);
+    const hexDigits = '0123456789ABCDEF';
+
+    var buffer = _buffer ??= Uint8List(16);
```

Comment:
Allocating typed data, and String.fromCharCodes, are both surprisingly expensive on the web.
A StringBuffer would likely be faster.

If the code was web-only, I think I would lean on the existing conversions and concatenate strings for the sub-parts.

    String result = (_m << 22 + _l).toRadixString(16);
    if (_h != 0) result = _h.toRadixString(16) + result.padLeft(11, '0');
    return result.toUpperCase();

Reply:
We really should split the class into native and web implementations. 
There are things where the a single best implementation simply does not exist.
(But maybe a string buffer and using singleton strings is reasonable. It just costs an up-to-64-bits pointer per digit, which seems excessive.)

Reply:
I can roll back this change, and take another go at it later. It just looked horribly inefficient as written, with the repeated string allocation giving quadratic behavior on native. (But it's what we have, so apparently it has not been a big problem yet.)